### ISSUE TITLE
Does not render static + dynamic children

### ISSUE THREAD

**Issue by jisaacks** [airbnb/enzyme#475]
I am doing full DOM rendering, when a component contains static OR dynamic children it works. However, when a component contains static AND dynamic children, it does not:

```javascript
let compA = mount(<div>hello</div>);
console.log('A', compA.debug());

let compB = mount(<div>{'world'}</div>);
console.log('B', compB.debug());

let compC = mount(<div>hello {'world'}</div>);
console.log('C', compC.debug());
```

__output:__
```
A <div>
  hello
</div>
B <div>
  world
</div>
C <div />
```

Notice how C is being redered as `<div />`. Expected: `<div>hello world</div>`.

Not sure if it is relevant but I am using jsdom.

**Comment by nfcampos** [comment]
```js
const w = mount(<div>hello {'world'}</div>)
console.log(w.debug()) // output: <div />
console.log(w.text()) // output: hello world
```
Looking into this a bit more I can tell you that while the children are being rendered, as the output of `.text()` shows, the `.debug()` method is failing to show the children in this case. I'll look into this more.

**Comment by nfcampos** [comment]
@jisaacks should be fixed by #476
